lib/npm.js

```diff
@@ -15,6 +16,42 @@ var Npmrc = require('./npmrc');
 
 var defaultRegistry = 'https://registry.npmjs.org';
 
+// Test whether the contents of buffer is gzipped
+function isGzip(buffer) {
+  if (!buffer || buffer.length < 3) {
+    return false;
+  }
+  return buffer[0] === 0x1f && buffer[1] === 0x8b && buffer[2] === 0x08;
+}
+
+// Return a promise that gunzips the buffer *if* it is gzipped
+// Otherwise return a promise with the buffer as is. 
+function gunzip(buffer) {
+  if (isGzip(buffer)) {
+    return new Promise(function (resolve, reject) {
+      zlib.gunzip(buffer, function (err, decoded) {
+        if (err) {
+          reject(err);
+        } else {
+          resolve(decoded);
+        }
+      });
+    });
+  } else {
+    return Promise.resolve(buffer);
+  }
+}
+
+// Create a readable stream from a buffer
+function bufferStream(buffer) {
+  return new stream.Readable({
+    read(size) {
+      this.push(buffer);
+      this.push(null);
+    }
```

Comment:
Ok, that sounds like it would be a better approach then.

We have a large memory overhead in jspm anyway, so yes we shouldn't add to that by buffering all archives into memory before writing.